cmake/CatkinBuild.cmake

```diff
@@ -0,0 +1,72 @@
+# Set compiler flags
+set(CMAKE_CXX_STANDARD 14)
+
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+if (NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+  add_definitions(-O3)
+endif(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+
+# Find catkin macros and libraries
+find_package(catkin REQUIRED)
+find_package(Eigen3 REQUIRED)
+find_package(Boost REQUIRED COMPONENTS python)
+
+find_package(OpenMP REQUIRED)
+if (OpenMP_FOUND)
+  add_compile_options("${OpenMP_CXX_FLAGS}")
+  add_definitions(-DHAVE_OPENMP=${OpenMP_FOUND})
+endif()
+
+# Catkin package macro
+catkin_package(
+  INCLUDE_DIRS
+    nabo
+    ${CMAKE_SOURCE_DIR}
+    ${EIGEN3_INCLUDE_DIR}
+  LIBRARIES
+    nabo
+  DEPENDS
+    Boost
+)
+
+########################
+## Library definition ##
+########################
+# Nabo
+add_library(nabo
+  ${NABO_SRC}
+  ${NABO_HEADERS}
+)
+
+target_include_directories(nabo
+  PUBLIC
+    ${CMAKE_SOURCE_DIR}
+    ${CMAKE_SOURCE_DIR}/nabo
+  SYSTEM
+    ${EIGEN3_INCLUDE_DIR}
+    ${Boost_INCLUDE_DIRS}
+    ${OpenMP_CXX_INCLUDE_DIRS}
+)
+
+target_link_libraries(nabo
+  ${catkin_LIBRARIES}
+  ${Boost_LIBRARIES}
+  ${OpenMP_CXX_LIBRARIES}
+)
+
+#############
+## Install ##
+#############
+install(
+  TARGETS
+    nabo
+  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+)
+
+install(
+  DIRECTORY
+    ${CMAKE_SOURCE_DIR}/nabo/
+  DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION}/nabo
```

Comment:
After testing your suggestion, I found that this breaks the catkin install, because libnabo has a very specific way of managing the include paths, intrinsic to its folder structure. If we only install the headers, the include directory is flattened out in the install location.

What I did instead was to add 

```
FILES_MATCHING
    PATTERN "*.h"
    PATTERN "*.hpp"
```

to the install clause.